final/main.py

```diff
@@ -0,0 +1,20 @@
+import utils
+from tsp import TSP
+
+population_size = 50
+n_generation = 100
+tournament_size = 20
+mutation_rate = 0.01 
+input_path = './tsp_data.txt'   
+
+def main():
```

Comment:
We can get inputs from "sys.argv". Of course, this is a suggestion.

Reply:
yes, you are true.
Done